botorch/models/kernels/contextual_lcea.py

```diff
@@ -61,7 +61,7 @@ def __init__(
         self.decomposition = decomposition
         self.batch_shape = batch_shape
         self.train_embedding = train_embedding
-        self.device = device
+        self._device = device
```

Comment:
Hmm I don't think we need to assign this anywhere, actually. If we just use the `device` kwarg to set things here in `__init__` then the parameter `raw_outputscale_list` will be set with the proper `device` here: https://github.com/pytorch/botorch/pull/1611/files#diff-3bc5cfbe1a19e8fd5c96a54e56c367f98e78171fe26b408794a67d641d27fc57R111-R122

If we then remove the `device` property and just use that of the base class, then the standard functionality 
should return the device correctly (note that `has_lenghtscale` is `False` here): https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/kernels/kernel.py#L265-L273

Reply:
That said, we'll have to do this for the `SACKernel` below, so feel free to keep the current version if you prefer.

Reply:
I went with this design for backwards compatibility: historically `self.device` has been `None` when the `device` argument isn't specified. That's probably not the ideal way for `self.device` to behave, but I figure we can reconsider that in a future PR.